sdk/src/main/java/com/moyasar/android/sdk/presentation/viewmodel/PaymentSheetViewModel.kt

```diff
@@ -0,0 +1,259 @@
+  private val createPaymentUseCase: CreatePaymentUseCase,
+  private val createTokenUseCase: CreateTokenUseCase,
+) : AndroidViewModel(application) {
+
+
+  private var ccOnChangeLocked = false
+  private var ccExpiryOnChangeLocked = false
+
+  private val _status =
+    MutableLiveData<PaymentStatusViewState>().default(PaymentStatusViewState.Reset)
+  private val _payment = MutableLiveData<Payment?>()
+
+  internal val payment: LiveData<Payment?> = _payment
+  val status: LiveData<PaymentStatusViewState> = _status
+  val isFormValid: LiveData<Boolean> = formValidator._isFormValid.distinctUntilChanged()
+
+
+  private val cleanCardNumber: String
+    get() = formValidator.number.value!!.replace(" ", "")
+
+  private val expiryMonth: String
+    get() = parseExpiry(formValidator.expiry.value ?: "")?.month.toString()
+
+  private val expiryYear: String
+    get() = parseExpiry(formValidator.expiry.value ?: "")?.year.toString()
+
+  // Done logic like this to replicate iOS SDK's behavior
+  val amountLabel: String
+    get() = getFormattedAmount(paymentConfig)
+
+  private fun notifyPaymentResult(paymentResult: PaymentResult) = callback(paymentResult)
+
+  /*************************
+   * Perform Create payment Request After submit button clicked and createSaveOnlyToken = false
+   ************************/
+  private fun createPayment() {
+    val request = PaymentRequest(
+      paymentConfig.amount,
+      paymentConfig.currency,
+      paymentConfig.description,
+      PaymentAuthFragment.RETURN_URL,
+      CardPaymentSource(
+        formValidator.name.value!!,
+        cleanCardNumber,
+        expiryMonth,
+        expiryYear,
+        formValidator.cvc.value!!,
+        if (paymentConfig.manual) "true" else "false",
+        if (paymentConfig.saveCard) "true" else "false",
+      ),
+      paymentConfig.metadata ?: HashMap()
+    )
+
+    scope(block = { createPaymentUseCase(request) }) { result ->
+      when (result) {
+        is RequestResultViewState.Success -> {
+          _payment.value = result.data
+
+          when (result.data.status.lowercase()) {
+            "initiated" -> {
+              _status.value =
+                PaymentStatusViewState.PaymentAuth3dSecure(result.data.getCardTransactionUrl())
+            }
+
+            else -> {
+              notifyPaymentResult(PaymentResult.Completed(result.data))
+            }
+          }
+        }
+
+        is RequestResultViewState.Failure -> {
+          notifyPaymentResult(PaymentResult.Failed(result.e))
+        }
+      }
+    }
+  }
+
+  /*************************
+   * Perform Create Save only Token Request After submit button clicked and createSaveOnlyToken = true
+   ************************/
+  private fun createSaveOnlyToken() {
+    val request = TokenRequest(
+      formValidator.name.value!!,
+      cleanCardNumber,
+      formValidator.cvc.value!!,
+      expiryMonth,
+      expiryYear,
+      true,
+      "https://sdk.moyasar.com"
```

Comment:
If I understand correctly, the value here should be the full RETURN_URL (`https://sdk.moyasar.com/payment/return`) . Can you please confirm?